### ISSUE TITLE
btrfs: ValueError: too many values to unpack

### ISSUE THREAD

**Issue by morph027** [seppovic/check_mk-plugins#2]
Agent output seems to look ok, but check_mk does not like it ;)

Agent output

```
<<<df_btrfs>>>
[btrfs_df]
/backup
Data, single: total=1.01GiB, used=687.91MiB
System, DUP: total=8.00MiB, used=16.00KiB
System, single: total=4.00MiB, used=0.00
Metadata, DUP: total=1.00GiB, used=944.00KiB
Metadata, single: total=8.00MiB, used=0.00
unknown, single: total=16.00MiB, used=0.00
Label: none  uuid: 364cb6f1-f08b-4d8a-ab4d-bc3b9b93cfe5
Total devices 1 FS bytes used 688.85MiB
devid    1 size 3.64TiB used 3.04GiB path /dev/mapper/backup_crypt

Btrfs v3.12
[df]
/dev/mapper/backup_crypt btrfs  3907015492 722728 3904184216       1% /backup
```

Check_MK debug

```
Check output:     check failed - please submit a crash report!
  Check_MK Version: 1.2.6p12
  Date:             2015-13-11 15:05:25
  Host:             backupkiste
  Service:          Filesystem /backup
  Check type:       df_btrfs
  Item:             '/backup'
  Parameters:       {'levels': (80.0, 90.0),
 'levels_low': (50.0, 60.0),
 'magic_normsize': 20,
 'trend_perfdata': True,
 'trend_range': 24}
  Traceback (most recent call last):
        File "/omd/sites/server/var/check_mk/precompiled/backupkiste", line 1115, in do_all_checks_on_host
'          \x00|\x0b\x00d\x01\x00\x15d\x0f\x00|\x01\x00k\x06\x00r\xce\x02|\x08\x00d\x13\x00\x17p\xd1\x02|\x08\x00|\x01\x00|\x08\x00\x19d\x16\x00d\x17\x00d\x18\x00d\x02\x00\x83\x03\x02\\\x03\x00}\x16\x00}\x17\x00}\x18\x00|\x17\x00r'
        File "/omd/sites/server/var/check_mk/precompiled/backupkiste", line 4286, in check_btrfs
        File "/omd/sites/server/var/check_mk/precompiled/backupkiste", line 2497, in df_check_filesystem_list
      ValueError: too many values to unpack 
```

**Comment by seppovic** [comment]
Hey morph, thx for your feedback.
on a first quick overview i cant find any serious differences in plugin output.
could you add a:

```
import pprint
pprint.pprint((item, params, fslist))
```

above line 235 (    (df_ret, output, perf) = df_check_filesystem_list(item, params, fslist)), run a 

```
check_mk -n -v -p --checks=df_btrfs backupkiste
```

in "server" sites context and paste the output here.

**Comment by morph027** [comment]
Hi!

Here you go:

```
Check_mk version 1.2.6p12
('/backup',
 {'levels': (80.0, 90.0),
  'levels_low': (50.0, 60.0),
  'magic_normsize': 20,
  'trend_perfdata': True,
  'trend_range': 24},
 [('/backup', 3815444.81640625, 3786218.30859375, 2002.91015625)])
Filesystem /backup   UNKNOWN - check failed - please submit a crash report!   ()
OK - Agent version 1.2.6p12, execution time 0.4 sec|execution_time=0.426 user_time=0.530 system_time=0.080 children_user_time=0.000 children_system_time=0.000
```

Thank you!

**Comment by seppovic** [comment]
just checked check_mk git repository and found [this](http://git.mathias-kettner.de/git/?p=check_mk.git;a=commitdiff;h=6161e1a7f5f1574a788f53bc5bc29df69d6ad2cd) commit in 1.2.7i3 branch. For a quick fix in 1.2.6 you could apply this:

```
--- df_btrfs    2015-11-16 08:07:51.641079000 +0100
+++ df_btrfs.1.2.6  2015-11-16 09:14:58.754951000 +0100
@@ -197,7 +197,7 @@
             avail_mb = properties["available"]
             size_mb = properties["size"]
             reserved_mb = size_mb - avail_mb - used_mb # reserved for root
-            fslist.append((mountpoint, size_mb, avail_mb, reserved_mb))
+            fslist.append((mountpoint, size_mb, avail_mb))

             # here we do the aktual logic, if a device is fully allocated
             # and Metadata is more than 70% full we raise an alarm
```

patch (untestet) and live without root reserve, which then may lead to false results because the root reserve is not taken into account when it comes to calculation of available discspace.
I will update the mkp's min required version of check_mk

**Comment by morph027** [comment]
Seems to work for the moment!

```
Check_mk version 1.2.6p12
('/backup',
 {'levels': (80.0, 90.0),
  'levels_low': (50.0, 60.0),
  'magic_normsize': 20,
  'trend_perfdata': True,
  'trend_range': 24},
 [('/backup', 3815444.81640625, 3758402.66796875)])
Filesystem /backup   OK - 1.50% used (55.71 GB of 3.64 TB), (levels at 80.00/90.00%), trend: 0.00 B / 24 hours (/backup=57042.1484375MB;3052355.853125;3433900.334766;0;3815444.816406 growth=0;;;; trend=0;;;0;158976.86735)
OK - Agent version 1.2.6p12, execution time 0.1 sec|execution_time=0.083 user_time=0.520 system_time=0.080 children_user_time=0.000 children_system_time=0.000
```
